tools/server/boot.js

```diff
@@ -129,10 +129,11 @@ Fiber(function () {
           return require(name);
         }
 
-        var nodeModuleDir =
-          path.resolve(serverDir, fileInfo.node_modules, name);
+        var nodeModuleBase = path.resolve(serverDir, fileInfo.node_modules);
 
-        if (fs.existsSync(nodeModuleDir)) {
+        var nodeModuleDir = path.resolve(nodeModuleBase, name);
+
+        if (fs.existsSync(path.resolve(nodeModuleBase, name.split("/")[0]))) {
```

Comment:
No, I think this looks like it could be right.  Isn't the point that `Npm.require("foo/bar")` should be resolved by saying "if this package uses `foo`, then we should be trying to resolve `"foo/bar"` as within this module's foo"?

Node require has a bunch of different options; for example, `require('foo/bar')` can pull in a file that's called `bar.js` inside the `foo` package.  Or that's called `bar/index.js` inside `foo`.